### ISSUE TITLE
TypeError: Cannot read property 'prototype' of undefined

### ISSUE THREAD

**Issue by alarrosa14** [drewzboto/grunt-connect-proxy#97]
I can't load grunt-connect-proxy. After installing it I get the following error after executing `grunt serve`:
```
Loading "connect_proxy.js" tasks...ERROR
>> TypeError: Cannot read property 'prototype' of undefined
```

Grunt version: 0.4.5

**Comment by thedug** [comment]
Same here. My build is blowing up.

Looks like the new version of eventemitter3 (1.0.1) is causing the error

Looking back through our build server this version passed..

grunt-connect-proxy@0.1.11 node_modules/grunt-connect-proxy
└── http-proxy@1.1.6 (eventemitter3@0.1.6)

This one failed

grunt-connect-proxy@0.1.11 node_modules/grunt-connect-proxy
└── http-proxy@1.1.6 (eventemitter3@1.0.1)

**Comment by thedug** [comment]
We are going to use npm shrinkwrap to force the older version of eventemitter3

**Comment by phobetron** [comment]
Here's the (closed) bug ticket for eventemitter3 https://github.com/primus/eventemitter3/issues/29

Looks like there are breaking changes that everyone is expected to patch against.

**Comment by lostthetrail** [comment]
I am also seeing this issue.

**Comment by sjerman** [comment]
shrinkwrapping does fix the issue...

**Comment by jaan** [comment]
same here:
```
$ grunt server --verbose
Initializing
Command-line options: --verbose
Reading "Gruntfile.js" Gruntfile...OK

Registering Gruntfile tasks.
Loading "Gruntfile.js" tasks...ERROR
>> TypeError: Cannot read property 'prototype' of undefined
>>     at Object.exports.inherits (util.js:634:43)
>>     at Object.<anonymous> (/ProjectPath/node_modules/grunt-connect-proxy/node_modules/http-proxy/lib/http-proxy/index.js:106:17)
>>     at Module._compile (module.js:460:26)
>>     at Object.Module._extensions..js (module.js:478:10)
>>     at Module.load (module.js:355:32)
>>     at Function.Module._load (module.js:310:12)
>>     at Module.require (module.js:365:17)
>>     at require (module.js:384:17)
>>     at Object.<anonymous> (/ProjectPath/node_modules/grunt-connect-proxy/node_modules/http-proxy/lib/http-proxy.js:4:17)
>>     at Module._compile (module.js:460:26)

**Comment by phobetron** [comment]
As a simple workaround, I pinned the eventemitter3 version in my package.json file.
```
...
"eventemitter3": "^0.1.6",
...
```

**Comment by emiller** [comment]
We also hit this issue. `http-proxy` released a new version ~9 hours ago which may be the likely cause. If @phobetron's simple fix works, then there is a good chance that the problematic commit in the new version is:

https://github.com/nodejitsu/node-http-proxy/commit/18c77cafc7d5479502cf5c4d2b663d8f85cfd6d4#diff-8bcb067b974b0e74fe86cdb07748315e

**Comment by thedug** [comment]
@phobetron How can you peg it in your package.json to fix this? This is a transitive dependency so it will install both versions right?

**Comment by thedug** [comment]
shrinkwrap fails for me on my build server. arg!

`linux-x64-node-0.10` exists; testing
Binary is fine; exiting
npm ERR! cb() never called!
npm ERR! not ok code 0

**Comment by phobetron** [comment]
All I know is, it worked for me. I placed that line among my dev dependencies.

**Comment by thedug** [comment]
Strange that did work. How does a project dependency impact a dependency from another module since these are downloaded into sub dirs.

**Comment by esseti** [comment]
have a look at this https://github.com/drewzboto/grunt-connect-proxy/issues/95

**Comment by ruiaraujo** [comment]
This issue has been solved in #96.
